src/libs/fileDownload/index.desktop.ts

```diff
@@ -0,0 +1,32 @@
+import ELECTRON_EVENTS from '@desktop/ELECTRON_EVENTS';
+import type {Options} from '@libs/downloadQueue/electronDownloadManagerType';
+import type {FileDownload} from './types';
+
+/**
+ * The function downloads an attachment on desktop platforms.
+ */
+const fileDownload: FileDownload = (url, fileName) => {
+    const options: Options = {
+        filename: fileName,
+        saveAs: true,
+    };
+    window.electron.send(ELECTRON_EVENTS.DOWNLOAD, {url, options});
+
+    return new Promise((resolve) => {
+        window.electron.on(ELECTRON_EVENTS.DOWNLOAD_COMPLETED, (args) => {
+            if (args.url !== url) {
+                return;
+            }
+            resolve();
+        });
+
+        window.electron.on(ELECTRON_EVENTS.DOWNLOAD_CANCELLED, (args) => {
+            if (args.url !== url) {
+                return;
+            }
+            resolve();
+        });
+    });
```

Comment:
This will block the browser process if you'd await on this. Why do we need to add these when we're resolving the promise any ways?

Reply:
If we need an event at any cost, we might add a `DOWNLOAD_STARTED` event instead. That would indicate that the download was started.

Reply:
Thanks, the code has been updated to only use `DOWNLOAD_STARTED`.